src/infer_stats.cc

```diff
@@ -36,10 +36,21 @@ namespace triton { namespace core {
 
 #ifdef TRITON_ENABLE_STATS
 
+// This function converts FailureReason enum values to std::string
+std::string failureReasonToString(FailureReason reason) {
+  switch (reason) {
+    case FailureReason::REJECTED: return "REJECTED";
```

Comment:
instead of introducing a separate data structure can we use the TRITON error types?

Reply:
note to @itskevinwang (who came by to my desk with some questions), let's focus on the following first:
- serve a model that loads successfully
- make sure the new failure metrics appear in output of `curl localhost:8002/metrics`
- have some tests that would test the expected logic - it's OK if the tests themselves fail (ex: expected failure count > 1, but got zero, or something like that)

Once you have server, model, and tests that can run together, we can iterate on the feedback more easily.